homeassistant/components/light/osramlightify.py

```diff
@@ -197,22 +236,71 @@ def turn_off(self, **kwargs):
             _LOGGER.debug("turn_off requested transition time for light:"
                           " %s is: %s ",
                           self._name, transition)
-            self._light.set_luminance(0, transition)
+            self._luminary.set_luminance(0, transition)
         else:
             transition = 0
             _LOGGER.debug("turn_off requested transition time for light:"
                           " %s is: %s ",
                           self._name, transition)
-            self._light.set_onoff(0)
-            self._state = self._light.on()
-
+        self._luminary.set_onoff(0)
+        self._state = False
         self.schedule_update_ha_state()
```

Comment:
You mean the `self._state = False` ?
Well yes I can remove it, but I saw, that without it, HA gets a lot more unresponsive, because otherwise it the GUI jumps back to the on state and then after a couple of seconds it jumps back to off. If that is okay, I'm willing to remove it, I just felt it works better with.